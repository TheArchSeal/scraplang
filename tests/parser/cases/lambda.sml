() => 0;
() => () => 0;
(x) => x;
(x, y, z) => x;
(x: i32) => x;
(x = 1) => x;
(x: i32 = 1) => x;
