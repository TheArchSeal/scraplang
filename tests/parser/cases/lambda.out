stmt (3):1:1 ;
    expr (5):1:1 ()=>
        expr (3):1:7 0
stmt (3):2:1 ;
    expr (5):2:1 ()=>
        expr (5):2:7 ()=>
            expr (3):2:13 0
stmt (3):3:1 ;
    expr (5):3:1 ()=>
        param   :3:2 x
        type (1):3:2 (inferred)
        expr (1):3:2 (empty)
        expr (3):3:8 x
stmt (3):4:1 ;
    expr (5):4:1 ()=>
        param   :4:2 x
        type (1):4:2 (inferred)
        expr (1):4:2 (empty)
        param   :4:5 y
        type (1):4:5 (inferred)
        expr (1):4:5 (empty)
        param   :4:8 z
        type (1):4:8 (inferred)
        expr (1):4:8 (empty)
        expr (3):4:14 x
stmt (3):5:1 ;
    expr (5):5:1 ()=>
        param   :5:2 x
        type (3):5:5 i32
        expr (1):5:2 (empty)
        expr (3):5:13 x
stmt (3):6:1 ;
    expr (5):6:1 ()=>
        param   :6:2 x
        type (1):6:2 (inferred)
        expr (3):6:6 1
        expr (3):6:12 x
stmt (3):7:1 ;
    expr (5):7:1 ()=>
        param   :7:2 x
        type (3):7:5 i32
        expr (3):7:11 1
        expr (3):7:17 x
